contrib/protobuf/template.py

```diff
@@ -0,0 +1,59 @@
+pkgname = "protobuf"
+pkgver = "21.12"
+pkgrel = 0
+build_style = "cmake"
+configure_args = [
+    "-DBUILD_SHARED_LIBS=ON",
+    "-Dprotobuf_USE_EXTERNAL_GTEST=ON",
+]
+hostmakedepends = ["cmake", "ninja", "pkgconf"]
+makedepends = ["zlib-devel", "gtest-devel"]
+pkgdesc = "Protocol buffers compiler"
+maintainer = "Jami Kettunen <jami.kettunen@protonmail.com>"
+license = "BSD-3-Clause"
+url = "https://protobuf.dev"
+source = f"https://github.com/protocolbuffers/protobuf/archive/v{pkgver}.tar.gz"
+sha256 = "22fdaf641b31655d4b2297f9981fa5203b2866f8332d3c6333f6b0107bb320de"
+# FIXME cfi makes protoc not compile any tests
+# FIXME vis causes JsonObjectWriterTest.EmptyRootObject test to fail with
+# Assertion failed: f == nullptr || dynamic_cast<To>(f) != nullptr (src/google/protobuf/stubs/casts.h: down_cast: 94)
+hardening = ["!vis", "!cfi"]
+
+if self.profile().cross:
+    hostmakedepends += ["protobuf"] # needs host protoc
+
+def post_install(self):
+    self.install_license("LICENSE")
+
+@subpackage("libprotobuf")
+def _libprotobuf(self):
+    self.pkgdesc = f"Protocol buffers C++ library"
+    return ["usr/lib/libprotobuf.so.*"]
+
+@subpackage("libprotobuf-lite")
+def _libprotobuf_lite(self):
+    self.pkgdesc = f"Protocol buffers C++ library (lite version)"
+    return ["usr/lib/libprotobuf-lite.so.*"]
+
+@subpackage("libprotoc")
+def _libprotoc(self):
+    self.pkgdesc = f"Protocol buffers compiler library"
+    return ["usr/lib/libprotoc.so.*"]
+
+@subpackage("libprotoc-devel")
+def _libprotoc_devel(self):
+    self.pkgdesc = f"Protocol buffers compiler library (development files)"
+    self.depends = [f"libprotoc~{pkgver}"]
+    return [
+        "usr/lib/libprotoc.so",
+        "usr/include/google/protobuf/compiler",
+    ]
+
+@subpackage("protobuf-devel")
+def _devel(self):
+    self.depends = [
+        f"libprotobuf~{pkgver}",
+        f"libprotobuf-lite~{pkgver}",
```

Comment:
technically this isn't needed, as the devel packages depend on the base libs already